src/pages/test-report/components/timer/timer.ts

```diff
@@ -0,0 +1,68 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'timer',
+  templateUrl: 'timer.html',
+})
+export class TimerComponent {
+
+  showStartTimerButton: boolean;
+
+  interval: NodeJS.Timeout;
+  seconds: number;
+  timerString: string;
+  touchTimeout: NodeJS.Timeout;
+  touchState: boolean = false;
+  touchStateDelay: number = 100;
+  isPaused: boolean = true;
+
+  constructor() {
+    this.showStartTimerButton = true;
+    this.seconds = 0;
+
+    this.generateTimerString();
+  }
+
+  toggleTimer = (): void => {
+    this.isPaused = !this.isPaused;
+    return this.interval ? this.pauseTimer() : this.startTimer();
+  }
+
+  startTimer = (): void => {
+    this.showStartTimerButton = false;
+    this.interval = setInterval(() => {
+      this.seconds += 1;
+      this.generateTimerString();
+    }, 1000);
+  }
+
+  pauseTimer = (): void => {
+    clearInterval(this.interval);
+    this.interval = undefined;
+  }
+
+  generateTimerString = (): void => {
+    const date: Date = new Date(1970, 0, 1 , 0 , 0, 0);
+    date.setSeconds(this.seconds);
+
+    const showExtraZeroHours = date.getHours() < 10;
+    const showExtraZeroMinutes = date.getMinutes() < 10;
+    const showExtraZeroSeconds = date.getSeconds() < 10;
+
+    this.timerString = `${
```

Comment:
Could we use use our `DateTime` provider to do some of the heavy lifting for us here?

Reply:
Could be, but I kinda like that it's a self-contained timer

Reply:
Nice and readable. I'd have been tempted to do something like `(date.getSeconds()+'').padStart(2,'0')` but the verboseness of this solution is preferable, definitely makes it more readable.